File: w09/d01/enoch_riese/mail/spec/package_spec.rb
require_relative "spec_helper"
require_relative "../lib/package"

describe Package do
  it "should have a content array" do
    pack = Package.new
    expect(pack.content).to be_instance_of(Array)
  end
  it "should have an is_fragile boolean" do
    pack = Package.new
    frag = pack.is_fragile
    expect(!!frag).to eq(frag)
  end
  it "should have an is_broken boolean" do
    pack = Package.new
    frag = pack.is_broken
    expect(!!frag).to eq(frag)
  end
  it "should have a method shake" do
    pack = Package.new
    expect(pack).to respond_to(:shake)
  end
  it "should break if fragile and shaken" do
    pack = Package.new
    pack.shake
    expect(pack.is_broken).to be_true
  end
  it "should have an is_broken method" do
    pack = Package.new
    expect(pack).to respond_to(:is_broken)
  end
end

File: w09/d03/jimmy_davis/mail_objects_app/spec/javascripts/package_spec.js
describe("a package", function(){

  var packageItem;

  beforeEach( function() {
    packageItem = new Package();
  });

  it("has a content array", function() {
    packageItem.addContent("shirt");
    packageItem.addContent("pants");
    expect(packageItem.content).toEqual(["shirt", "pants"]);
  });

  it("has a fragile boolean", function(){
    packageItem.isFragile(true);
    expect(packageItem.fragile).toBe(true);
    var test = function() {
      packageItem.isFragile("not");
    };
    expect(test).toThrow();
  });

  it("has a broken boolean", function() {
    expect(packageItem.broken).toBe(false);
  });

  it("has a shake method", function() {
    packageItem.shake();
    expect(packageItem.broken).toBe(false);
    packageItem.isFragile(true);
    packageItem.shake();
    expect(packageItem.broken).toBe(true);
  });

  it("has an isBroken method that returns whether broken or not", function() {
    expect(packageItem.isBroken()).toBe(false);
    packageItem.isFragile(true);
    packageItem.shake();
    expect(packageItem.isBroken()).toBe(true);
  });

});
